src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final int DATE_INDEX = 0;
+    private static final int HOURS_INDEX = 2;
+    private static final int PAY_FOR_HOUR_INDEX = 3;
+    private static final DateTimeFormatter dateTimeFormatter =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data,
+                                String dateFrom, String dateTo) {
+        LocalDate parseDateFrom = LocalDate.parse(dateFrom, dateTimeFormatter);
+        LocalDate parseDateTo = LocalDate.parse(dateTo, dateTimeFormatter);
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (String name : names) {
+            int salary = 0;
+            for (String element : data) {
+                LocalDate date = LocalDate.parse(element.split(" ")[DATE_INDEX],
+                        dateTimeFormatter);
+                String[] splitData = element.split(" ");
```

Comment:
```suggestion
                String[] splitData = element.split(" ");
                LocalDate date = LocalDate.parse(splitData[DATE_INDEX],
                        dateTimeFormatter);
```
you are splitting line twice